lib/parser/messages.rb

```diff
@@ -58,6 +58,7 @@ module Parser
     :masgn_as_condition      => 'multiple assignment in conditional context',
     :block_given_to_yield    => 'block given to yield',
     :invalid_regexp          => '%{message}',
+    :invalid_return          => 'Invalid return in class/module body',
```

Comment:
`'return cannot be used in class or module body'`

Reply:
This message was copied from MRI, I thought parser should match them.

Reply:
Not really, we try to make our messages readable. The only thing that directly matches MRI is *when* they're emitted, not *what*. MRI does not have particularly good ones.

Reply:
I mean, I understand how `@def_level` works (and it looks similar to the `Context`), the difference is *how* parser uses it. `in_def?` checks for `@def_level > 0` (i.e. that parser is in the `def` context), so the difference comes in the following code:
``` ruby
def m
  class << something
    # right here, @def_level is 1, so in_def? returns true
    # but Context#in_def? returns false,
    # because current context is singleton class (in_sclass? returns true)
  end
end
```
So we need to differentiate these methods somehow:
1. a method that checks that the parser is *directly* in some context (like `directly_in_def?` that returns `@stack.last == :def`) 
2. a method that checks that the parser is *somewhere* in the context, but not necessary *directly* in this context (like `somewhere_in_def?` that returns `@stack.include?(:def)`)

Hope it makes sense 😄

Reply:
`in_def?` and `indirectly_in_def?` sound good to me, and the overall concept too. Thanks for being patient with me!